src/components/EmojiPicker/EmojiPickerMenu/index.js

```diff
@@ -85,15 +76,40 @@ function EmojiPickerMenu({forwardedRef, onEmojiSelected}) {
         setArePointerEventsDisabled(false);
     }, [arePointerEventsDisabled]);
 
-    /**
-     * Focuses the search Input and has the text selected
-     */
-    function focusInputWithTextSelect() {
-        if (!searchInputRef.current) {
-            return;
-        }
-        searchInputRef.current.focus();
-    }
+    const onFocusedIndexChange = useCallback(
+        (newIndex) => {
+            if (filteredEmojis.length === 0) {
+                return;
+            }
+
+            if (highlightFirstEmoji) {
+                setHighlightFirstEmoji(false);
+            }
+
+            if (!isUsingKeyboardMovement) {
+                setIsUsingKeyboardMovement(true);
+            }
+            if (!arePointerEventsDisabled) {
+                setArePointerEventsDisabled(true);
+            }
+
+            // If the input is not focused and the new index is out of range, focus the input
+            if (newIndex < 0 && !searchInputRef.current.isFocused()) {
+                searchInputRef.current.focus();
+            }
+        },
+        [arePointerEventsDisabled, filteredEmojis.length, highlightFirstEmoji, isUsingKeyboardMovement],
+    );
+
+    const [focusedIndex, setFocusedIndex] = useArrowKeyFocusManager({
+        maxIndex: filteredEmojis.length - 1,
+        // Spacers indexes need to be disabled so that the arrow keys don't focus them. All headers are hidden when list is filtered
+        disabledIndexes: isListFiltered ? [] : [...headerIndices, ...spacersIndexes],
```

Comment:
this recreates the array with each render, are you able to profile this part of it doesn't perform bad?

Reply:
moved this into useMemo as spacer indexes might change when new emoji is added into frequently used emojis